### ISSUE TITLE
Synching i18n locale changes can break content edit forms

### ISSUE THREAD

**Issue by semiaddict** [boazpoolman/strapi-plugin-config-sync#8]
## Bug description
This is a follow up on the discussion that started in PR https://github.com/boazpoolman/strapi-plugin-config-sync/pull/7
When using the i8n plugin, an installation can become unusable if the default locale is deleted and replaced by another before performing a sync.

### Steps to reproduce the behavior
1. Setup  two strapi apps (see docker-compose example below)
2. Update both installations to 3.6.1
3. Add i18n and config-sync plugin in both installations
4. Create test content type in both with same options and fields
5. Export configs in first and import in second installation
6. Add some content in second installation
7. add FR locale under admin/settings/internationalization in first installation
8. set FR locale as default in first installation
9. delete EN locale in first installation
10. Export configs in first installation and import in second installation
11. Got to edit a content in second installation
12. "Something went wrong." shows up instead of the edit form 

### Expected behavior
The edit form allows editing the content as usual.

### Screenshots
Error page:
![image](https://user-images.githubusercontent.com/522064/117684460-4cc35a00-b1b5-11eb-94e7-42284c11105f.png)

### Code snippets
Sample docker-compose file used to reproduce the bug:
```yml
version: '3'
services:
  strapi:
    image: strapi/strapi
    environment:
      DATABASE_CLIENT: mongo
      DATABASE_NAME: strapi
      DATABASE_HOST: mongo
      DATABASE_PORT: 27017
      DATABASE_USERNAME: strapi
      DATABASE_PASSWORD: strapi
    volumes:
      - ./app:/srv/app
    ports:
      - 1337:1337
    depends_on:
      - mongo

  mongo:
    image: mongo
    environment:
      MONGO_INITDB_DATABASE: strapi
      MONGO_INITDB_ROOT_USERNAME: strapi
      MONGO_INITDB_ROOT_PASSWORD: strapi
    volumes:
      - ./.data:/data/db

  strapi2:
    image: strapi/strapi
    environment:
      DATABASE_CLIENT: mongo
      DATABASE_NAME: strapi
      DATABASE_HOST: mongo2
      DATABASE_PORT: 27017
      DATABASE_USERNAME: strapi
      DATABASE_PASSWORD: strapi
    volumes:
      - ./app2:/srv/app
    ports:
      - 1338:1337
    depends_on:
      - mongo2

  mongo2:
    image: mongo
    environment:
      MONGO_INITDB_DATABASE: strapi
      MONGO_INITDB_ROOT_USERNAME: strapi
      MONGO_INITDB_ROOT_PASSWORD: strapi
    volumes:
      - ./.data2:/data/db
```

## System
Node.js version: 14.16.0
NPM version: 7.7.6
Strapi version: 3.6.1
Database: MongoDB
Operating system: Docker

## Additional context
The issua can be manually fixed by making the same locale changes in the second installation (see steps 7-9).

**Comment by boazpoolman** [comment]
Thank you @semiaddict for the detailed issue submission!
Based on your description it seems most likely this issue would have been prevented if the i18n locales would be exported/imported as a new config type. I'd suggest we call this new config type `i18n-locale`.
I'll look into this!

**Comment by semiaddict** [comment]
Hello @boazpoolman,

Thank you for your work.
I just tested with the procedure listed in the issue description.
The issue seems to be resolved, as I can access the content edit form, make changes to existing content, and add new content.

However, locales continue to show up under admin/plugins/config-sync after imports, and the deleted locale in the first installation is not being deleted in the second installation upon syncing.
Here are a few screenshots from the second installation after having performed the sync of step 10:

![image](https://user-images.githubusercontent.com/522064/120769009-fd0a5100-c51c-11eb-80fc-18fa6e3e4802.png)

![image](https://user-images.githubusercontent.com/522064/120769109-11e6e480-c51d-11eb-90f8-46d11066a610.png)

![image](https://user-images.githubusercontent.com/522064/120769147-1ad7b600-c51d-11eb-9ed6-fa52f83753ec.png)

Those issue are much less problematic than the original one, as they do not block the use of the installation.

Thank you again for all your work.
Best.

**Comment by boazpoolman** [comment]
It's been a while, and a lot has changed.
Strapi v4 has been released, and this plugin was migrated along with it.

In the migrated plugin the `i18n-locale` config was added as a config type.
Also the `createdAt` and `updatedAt` field are excluded from the sync making the plugin a lot more useable.

Released in [v1.0.0-alpha.1](https://github.com/boazpoolman/strapi-plugin-config-sync/releases/tag/1.0.0-alpha.1) of `strapi-plugin-config-sync`.
